test/hexagon/include/halide-hexagon-setup.h

```diff
@@ -72,3 +75,27 @@ Expr sat_8(Expr e) {
 Expr usat_8(Expr e) {
   return clamp(e, 0, 255);
 }
+
+Expr saturating_subtract(Expr a, Expr b) {
+  Type wider = Int(a.type().bits() * 2);
+  /* FIXME: There is a problem here that I do not already know how to address.
+    There is no way to saturate and pack shorts (signed or unsigned) into signed
+    bytes. Assuming 'a' and 'b' were, say, i8x64 types, we'd be doing exactly
+    this, i.e. saturating and packing the widened result of the subtract
+    (i16x64) into signed bytes. This isn't supported. Should we be warning here?
```

Comment:
Thanks for the info. Can you want to make the associated changes in this branch? It sounds like maybe there's a warning/error that needs to be removed from CodeGen_Hexagon?